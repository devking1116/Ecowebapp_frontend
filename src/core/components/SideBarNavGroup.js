/* @flow */
import React from 'react';
import { Menu } from 'semantic-ui-react';

type Props = {
  isExpanded?: bool,
  children?: any,
}

/**
 * A group of <SideBarNavItem />s, for side bars used for navigation side bars.
 */
export default function SideBarNavGroup({ isExpanded, children, ...otherProps }: Props): React.Element<*> {
  return (
    <Menu.Menu {...otherProps}>
      {React.Children.map(children, (child) =>
        <Menu.Item>
          {React.cloneElement(child, { isExpanded })}
        </Menu.Item>
      )}
    </Menu.Menu>
  );
}
