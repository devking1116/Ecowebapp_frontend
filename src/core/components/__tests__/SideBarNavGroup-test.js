import React from 'react';
import renderer from 'react-test-renderer';

import SideBarNavGroup from '../SideBarNavGroup';

test('when expanded, it displays its children in a <Menu.Menu />.', () => {
  const component = renderer.create(
    <SideBarNavGroup expandedWidth={200} isExpanded>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SideBarNavGroup>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when not expaned, it displays its children in a <Menu.Menu />.', () => {
  const component = renderer.create(
    <SideBarNavGroup expandedWidth={200} >
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SideBarNavGroup>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when expanded, it gives its children the isExpanded prop.', () => {
  const component = renderer.create(
    <SideBarNavGroup isExpanded>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SideBarNavGroup>
  );
  const tree = component.toJSON();
  const { children: [ { children: [ header ] }, ...childrenMenuItems ] } = tree;

  expect(header.props.isExpanded).toBe(true);

  childrenMenuItems.forEach(({ children: [ child ]}) => {
    expect(child.props.isExpanded).toBe(true);
  });
});

test('when not expanded, it does not gives its children the isExpanded prop.', () => {
  const component = renderer.create(
    <SideBarNavGroup>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SideBarNavGroup>
  );
  const tree = component.toJSON();const { children: [ header, ...children ] } = tree;

  expect(header.props.isExpanded).toBeFalsy();

  children.forEach(({ children: [ child ]}) => {
    expect(child.props.isExpanded).toBeFalsy();
  });
});
