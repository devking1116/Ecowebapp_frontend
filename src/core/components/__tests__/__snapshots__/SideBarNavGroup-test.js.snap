exports[`test when expanded, it displays its children in a <Menu.Menu />. 1`] = `
<div
  className="menu"
  expandedWidth={200}>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 0
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 1
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 2
    </i>
  </div>
</div>
`;

exports[`test when not expaned, it displays its children in a <Menu.Menu />. 1`] = `
<div
  className="menu"
  expandedWidth={200}>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={undefined}>
      Test Child 0
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={undefined}>
      Test Child 1
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={undefined}>
      Test Child 2
    </i>
  </div>
</div>
`;
